bicep/deploy/resources.bicep

```diff
@@ -0,0 +1,50 @@
+targetScope = 'resourceGroup'
+
+@allowed([
+  'DevTest'
+  'Production'
+])
+@maxLength(10)
+@description('The environment type')
+param environmentType string = 'DevTest'
+
+@description('Optional. Location for resource.')
+param location string = resourceGroup().location
+
+@description('The resource name')
+param resourceName string
+
+@description('Metadata Tags for the resouce')
+param tags object
+
+var environmentConfig = loadJsonContent('config-environment.json')['${environmentType}']
+
+module storageAccount './modules/storageAccount.bicep' = {
+  name: 'StorageAccount'
+  params: {
+    location: location
+    resourceName: resourceName
```

Comment:
Do we want a simplified resourceName somewhere or maybe directly use toLower and/or replace("-", "")?

I can see the current naming convention used in main does not use dashes, but this will still fail if people use capital letters in company or system abbreviation

Reply:
Hmm good question. Generally I dislike code, which has "unknown" side-effects.

We could expand on line 32 in main.bicep, and set all 3 parameters toLower()?
It would be this place one would adjust the naming scheme anyhow to fit a specific use-case and/or customer naming scheme.